include/Platform/template.h

```diff
@@ -0,0 +1,611 @@
+  uint32_t ovrlap =
+      uint32_t(ceil(overlap * float(alig->originalNumberOfSequences - 1)));
+
+
+
+  // Depending on alignment type, indetermination symbol will be one or other
+  char indet = (alig->getAlignmentType() & SequenceTypes::AA) ? 'X' : 'N';
+
+  // prepare constant SIMD vectors
+  const Vector ALLINDET = Vector::duplicate(indet);
+  const Vector ALLGAP = Vector::duplicate('-');
+  const Vector ONES = Vector::duplicate(1);
+
+  // allocate aligned memory for faster SIMD loads
+  uint32_t *hits =
+      aligned_array<uint32_t, Vector>(alig->originalNumberOfResidues);
+  uint8_t *hits_u8 =
+      aligned_array<uint8_t, Vector>(alig->originalNumberOfResidues);
+
+  // for each sequence in the alignment, computes its overlap
+  for (int i = 0; i < alig->originalNumberOfSequences; i++) {
+
+    // reset hits count
+    memset(&hits[0], 0, alig->originalNumberOfResidues * sizeof(uint32_t));
+    memset(&hits_u8[0], 0, alig->originalNumberOfResidues * sizeof(uint8_t));
+
+    const uint8_t *datai =
+        reinterpret_cast<const uint8_t *>(alig->sequences[i].data());
+
+    // compare sequence to other sequences for every position
+    for (int j = 0; j < alig->originalNumberOfSequences; j++) {
+
+      // don't compare sequence to itself
+      if (j == i)
+        continue;
+
+      const uint8_t *dataj =
+          reinterpret_cast<const uint8_t *>(alig->sequences[j].data());
+
+      int k = 0;
+
+      // run iterations in SIMD while possible
+      for (; ((int)(k + Vector::LANES)) <= alig->originalNumberOfResidues;
+           k += Vector::LANES) {
+        // load data for the sequences
+        const Vector seqi = Vector::loadu(&datai[k]);
+        const Vector seqj = Vector::loadu(&dataj[k]);
+        // find which sequence characters are gap or indet
+        const Vector gapsi = (seqi == ALLGAP) | (seqi == ALLINDET);
+        const Vector gapsj = (seqj == ALLGAP) | (seqj == ALLINDET);
+        const Vector gaps = !(gapsi | gapsj);
+        // find which sequence characters match
+        const Vector eq = (seqi == seqj);
+        // find position where either not both characters are gap, or they are
+        // equal
+        const Vector n = (eq | gaps) & ONES;
+        // update counters
+        Vector hit = Vector::load(&hits_u8[k]);
+        hit += n;
+        hit.store(&hits_u8[k]);
+      }
+
+      // process the tail elements when there remain less than
+      // can be fitted in a SIMD vector
+      for (; k < alig->originalNumberOfResidues; k++) {
+        int nongapi = (datai[k] != indet) && (datai[k] != '-');
+        int nongapj = (dataj[k] != indet) && (dataj[k] != '-');
+        hits_u8[k] += ((nongapi && nongapj) || (datai[k] == dataj[k]));
+      }
+
+      // we can process up to UCHAR_MAX sequences, otherwise hits_u8[k]
+      // may overflow, so every UCHAR_MAX iterations we transfer the
+      // partial hit counts from `hits_u8` to `hits`
+      if ((j % UCHAR_MAX) == 0) {
```

Comment:
If the sequence to be compared (i) is a multiple of UCHAR_MAX hits_u8 does not get reset. I add comments with the code consider this and the rest of cases.